lib/manipulate_file.rb

```diff
@@ -1,17 +1,38 @@
+require 'json'
+
 class ManipulateFile
     
-  attr_reader :file
+  attr_reader :file, :name
 
   def initialize(file)
     begin
-      @file = File.readlines(file, chomp: true)
+        @file = File.readlines(file, chomp: true)
+        @name = File.basename(file)
     rescue Errno::ENOENT
       raise "File doesn't exist."
     end
   end
 
   def get_first_line
-    self.file[0]
+    @file[0]
+  end
+
+  def get_players_name
+    players = []
+    for element in @file do
+      if element.include?('ClientUserinfoChanged:')
+        player = element.split("\\")
+        if not players.include?(player[1])
+          players.append(player[1])
+        end
+      end
+    end
+    return players
+  end
+
+  def mount_json
+    obj = { @name => { "lines" => @file.length, "players" => self.get_players_name} }
```

Comment:
```suggestion
    obj = { @name => { "lines" => @file.length, "players" => get_players_name} }
```
Não precisa especificar o self aqui pois o ruby sabe que esse método é interno